@utahdts/utah-design-system/react/components/forms/FileInput/FileInput.jsx

```diff
@@ -0,0 +1,171 @@
+    const fileIndex = currentFiles.findIndex((item) => isEqual(file, item));
+    if (fileIndex !== -1) {
+      currentFiles.splice(fileIndex, 1);
+      // Create new FileList
+      const dataTransfer = new DataTransfer();
+      currentFiles.forEach((item) => dataTransfer.items.add(item));
+      // @ts-ignore
+      inputRef.current.files = dataTransfer.files;
+      currentOnChange();
+    }
+  }, []);
+
+  useEffect(() => {
+    if (files.length) {
+      addPoliteMessage(`You have selected the file: ${[...files].map((item) => item.name).join(', ')}.`);
+    } else {
+      addPoliteMessage('No file selected.');
+    }
+  }, [files]);
+
+  useEffect(() => {
+    // Track files when controlled
+    if (onChange) { setFiles(value || []); }
+  }, [value, onChange]);
+
+  return (
+    <div className="input-wrapper" ref={innerRef}>
+      <label htmlFor={id}>
+        {label}
+        {isRequired ? <RequiredStar /> : null}
+      </label>
+      {hint
+        ? (
+          <div className="info-box file-input__info-box mt-spacing-xs">
+            <div className="info-box__content">
+              {hint}
+            </div>
+          </div>
+        )
+        : null}
+      <div className={joinClassNames(
+        'file-input__box mt-spacing-s',
+        isDragged ? 'file-input__box--dragged' : '',
+        isDisabled ? 'file-input__box--disabled' : ''
+      )}
+      >
+        <div className="file-input__safari" />
+        {!files.length
+          ? (
+            <div className="file-input__instructions">
+              <span className="button button--small">Drag file{multiple ? 's' : ''} here or click to upload</span>
+            </div>
+          )
+          : ''}
+        <input
+          accept={acceptedFileTypes}
+          aria-describedby={errorMessage ? `${id}-error` : undefined}
+          className={className}
+          disabled={isDisabled}
+          id={id}
+          multiple={multiple}
+          name={name || id}
+          onChange={currentOnChange}
+          onDragEnter={() => setDragged(true)}
+          onDragLeave={() => setDragged(false)}
+          onDrop={() => setDragged(false)}
+          ref={inputRef}
+          type="file"
+        />
+        {files.length
+          ? (
+            <div className="file-input__file-selected">
+              <div className="flex justify-between items-center">
+                <span className="font-bold mr-spacing">{files.length} file{files.length > 1 ? 's' : ''} selected</span>
+                <span className="button button--small">Change file{files.length > 1 ? 's' : ''}</span>
+              </div>
+              <hr />
+              <div className="file-input__file-list flex-wrap">
+                {[...files].map((/** @type {File} */ file) => {
+                  if (children && isFunction(children)) {
+                    return (
+                    // @ts-ignore
+                    // eslint-disable-next-line react/jsx-no-constructed-context-values
```

Comment:
does this need to use a context? file and removeFile could be passed directly in to child:
`{children(file, removeFile)}`

Is it possible that file.name could have duplicates? If the file.name does not include the full path then two files with the same name from different locations could cause the key to not be unique.

Reply:
{children(file, removeFile)}